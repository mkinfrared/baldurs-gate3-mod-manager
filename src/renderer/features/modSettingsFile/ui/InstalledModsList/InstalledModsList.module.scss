.InstalledModsList {
  .table {
    width: 100%;

    th:first-child,
    td:first-child {
      width: 50px;
    }
  }
}
