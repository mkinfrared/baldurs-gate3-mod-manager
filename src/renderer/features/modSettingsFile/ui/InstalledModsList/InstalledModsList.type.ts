export type InstalledModsListProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
};
