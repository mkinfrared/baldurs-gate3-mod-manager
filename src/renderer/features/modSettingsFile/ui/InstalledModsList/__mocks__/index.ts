export { default } from "./InstalledModsList.mock";
