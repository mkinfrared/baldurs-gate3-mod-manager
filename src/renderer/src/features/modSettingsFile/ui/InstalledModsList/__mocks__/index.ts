export { InstalledModsListMock as InstalledModsList } from "./InstalledModsList.mock";
