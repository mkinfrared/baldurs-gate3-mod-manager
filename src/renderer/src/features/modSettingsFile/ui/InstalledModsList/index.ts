export { InstalledModsList } from "./InstalledModsList";

export * from "./InstalledModsList.type";
