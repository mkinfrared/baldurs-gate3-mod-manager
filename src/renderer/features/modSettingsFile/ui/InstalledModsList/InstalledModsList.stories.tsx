import { Meta, StoryObj } from "@storybook/react";

import { InstalledModsList } from "./InstalledModsList";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "ENTITIES/modSettingsFile/InstalledModsList",
  component: InstalledModsList,
  tags: ["autodocs"],
} satisfies Meta<typeof InstalledModsList>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <InstalledModsList {...args} />
    </div>
  ),
};

export { Default };

export default meta;
