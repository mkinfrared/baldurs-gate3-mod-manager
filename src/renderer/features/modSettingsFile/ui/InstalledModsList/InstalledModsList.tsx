import { InstalledMod } from "renderer/features/modSettingsFile/ui/InstalledMod";
import { classNames, trpc } from "renderer/shared/lib/helpers";
import { Heading } from "renderer/shared/ui";

import { EmptyList } from "../EmptyList";

import css from "./InstalledModsList.module.scss";
import { InstalledModsListProps } from "./InstalledModsList.type";

// TODO fix when getInstalledMods is refactored
// const installedMods = suspend(getInstalledMods);
const InstalledModsList = ({ className }: InstalledModsListProps) => {
  const { data } = trpc.mod.getInstalledMods.useQuery();
  const activeMods = data?.filter(({ isActive }) => isActive);

  return (
    <div
      className={classNames(css.InstalledModsList, className)}
      data-testid="InstalledModsList"
    >
      <div>
        <Heading variant="h3">Active Mods</Heading>
        {activeMods?.length ? (
          activeMods.map((mod) => (
            <InstalledMod key={mod.uuid ?? mod.name} mod={mod} />
          ))
        ) : (
          <EmptyList>
            <Heading variant="h4">
              Activate the mods from the Installed Mods list below using
              Activate button and they will appear here
            </Heading>
          </EmptyList>
        )}
      </div>
      <div>
        <Heading variant="h3">Installed Mods</Heading>
        {data?.length ? (
          data?.map((mod) => (
            <InstalledMod key={mod.uuid ?? mod.name} mod={mod} />
          ))
        ) : (
          <EmptyList>
            <Heading variant="h4">
              You haven't installed any mods yet. Use the panel on the right to
              install mods
            </Heading>
          </EmptyList>
        )}
      </div>
    </div>
  );
};

export { InstalledModsList };
