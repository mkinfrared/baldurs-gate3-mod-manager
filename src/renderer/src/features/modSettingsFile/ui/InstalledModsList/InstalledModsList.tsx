import { InstalledMod } from "@renderer/features/modSettingsFile/ui/InstalledMod";
import { ModsOrder } from "@renderer/features/modSettingsFile/ui/ModsOrder";
import { classNames, trpc } from "@renderer/shared/lib/helpers";
import { Heading, StickyBlock } from "@renderer/shared/ui";

import { EmptyList } from "../EmptyList";

import css from "./InstalledModsList.module.scss";
import { InstalledModsListProps } from "./InstalledModsList.type";

const InstalledModsList = ({ className }: InstalledModsListProps) => {
  const { data } = trpc.mod.getInstalledMods.useQuery();

  return (
    <div
      className={classNames(css.InstalledModsList, className)}
      data-testid="InstalledModsList"
    >
      <ModsOrder mods={data?.activeMods} />
      <div>
        <StickyBlock>
          <Heading variant="h3">Installed Mods</Heading>
        </StickyBlock>
        {data?.installedMods.length ? (
          data?.installedMods.map((mod, index) => (
            <InstalledMod
              key={mod.uuid ?? mod.name}
              mod={mod}
              position={index + 1}
            />
          ))
        ) : (
          <EmptyList>
            <Heading variant="h4">
              You haven't installed any mods yet. Use the panel on the right to
              install mods
            </Heading>
          </EmptyList>
        )}
      </div>
    </div>
  );
};

export { InstalledModsList };
