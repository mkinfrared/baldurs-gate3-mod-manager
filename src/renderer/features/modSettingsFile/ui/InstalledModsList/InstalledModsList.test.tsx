import { render } from "@testing-library/react";

import { InstalledModsList } from "./InstalledModsList";

describe("<InstalledModsList />", () => {
  const Component = <InstalledModsList />;

  it("should be defined", () => {
    expect(InstalledModsList).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("InstalledModsList");

    expect(element).toBeDefined();
  });
});
