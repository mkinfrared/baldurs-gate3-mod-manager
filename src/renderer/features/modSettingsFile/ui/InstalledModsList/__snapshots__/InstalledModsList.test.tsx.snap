// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InstalledModsList /> should match the snapshot 1`] = `
<div>
  <div
    class="InstalledModsList"
    data-testid="InstalledModsList"
  >
    <table
      class="Table DataTable table"
      data-testid="Table"
    >
      <thead>
        <tr>
          <th>
            Active
          </th>
          <th>
            Name
          </th>
          <th>
            Version
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
